x-pack/plugins/security_solution/public/detections/components/rules/eql_query_bar/eql_query_bar.tsx

```diff
@@ -52,9 +70,15 @@ export const EqlQueryBar: FC<EqlQueryBarProps> = ({
 }) => {
   const { addError } = useAppToasts();
   const [errorMessages, setErrorMessages] = useState<string[]>([]);
-  const { isValidating, setValue } = field;
+  const {
+    isValidating,
+    value: fieldValue,
+    setValue: setFieldValue,
+  } = field as FieldHook<FieldValueQueryBar>;
   const { isValid, message, messages, error } = getValidationResults(field);
-  const fieldValue = field.value.query.query as string;
+
+  const { uiSettings } = useKibana().services;
+  const [filterManager] = useState<FilterManager>(new FilterManager(uiSettings));
```

Comment:
nit: looks like we can use `useRef` here as `filterManager` won't be changed once set.

Reply:
done